verifier/truststore/truststore.go

```diff
@@ -89,40 +87,35 @@ func (trustStore *x509TrustStore) GetCertificates(ctx context.Context, storeType
 		if err != nil {
 			return nil, fmt.Errorf("error while reading certificates from %q: %w", joinedPath, err)
 		}
-
-		if err := validateCerts(certs, joinedPath); err != nil {
+		if err := ValidateCerts(certs, storeType, namedStore); err != nil {
```

Comment:
How about

```go
func ValidateCertificateChain(certs []*x509.Certificate) error {
	if len(certs) < 1 {
		return errors.New("empty certificate chain")
	}

	for _, cert := range certs {
		if !cert.IsCA {
			if err := cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature); err != nil {
				return fmt.Errorf(
					"certificate with subject %q is not a CA certificate or self-signed signing certificate",
					cert.Subject,
				)
			}
		}
	}

	return nil
}
```

```suggestion
		// to prevent any trust store misconfigurations, ensure there is at least
		// one certificate from each file.
		if len(certs) < 1 {
			return fmt.Errorf("%s: could not parse a certificate from. every file in a trust store must have a PEM or DER certificate in it", joinedPath)
		}
		if err := ValidateCertificateChain(certs); err != nil {
			return fmt.Errorf("%s: %w", joinedPath, err)
		}
```